src/main/java/esa/sen2vm/input/gipp/GIPPFileManager.java

```diff
@@ -0,0 +1,169 @@
+import java.util.regex.Pattern;
+
+/**
+ * Manager for gipp files
+ *
+ */
+public class GIPPFileManager {
+
+    /*
+     * Get sen2VM logger
+     */
+    private static final Logger LOGGER = Logger.getLogger(GIPPFileManager.class.getName());
+
+    /**
+     * List of GIPP Viewing Direction XML files
+     */
+    protected List<File> viewingDirectionFileList = new ArrayList<>();
+
+    /**
+     * GIPP Blind pixel XML files
+     */
+    protected File blindPixelFile = null;
+
+    /**
+     * GIPP spa mod XML files
+     */
+    protected File spaModFile = null;
+
+    /**
+     * Default constructor
+     */
+    public GIPPFileManager() {
+
+    }
+
+    /**
+     * Get GIPP file of a specific type by passing a regex
+     * @param folders that may
+     * @param a regex that match a specific type of GIPP
+     * @return an array that contains all files that correspond to the input regex
+     */
+    public List<File> findFiles(File[] directories, String regexMatch) {
+        List<File> foundFiles = new ArrayList<>();
+        Pattern pattern = Pattern.compile(regexMatch);
+
+        if (directories != null) {
+            for (File dir : directories) {
+                if (dir.isDirectory()) {
+                    File[] files = dir.listFiles(new FilenameFilter() {
+                        @Override
+                        public boolean accept(File dir, String name) {
+                            Matcher matcher = pattern.matcher(name);
+                            return matcher.matches();
+                        }
+                    });
+
+                    if (files != null) {
+                        for (File file : files) {
+                            foundFiles.add(file);
+                        }
+                    }
+                }
+            }
+        }
+
+        if (foundFiles != null && foundFiles.size() > 0) {
+            return foundFiles;
+        }
+        else {
+            return null;
+        }
+    }
+
+    /**
+     * Get GIPP file of a specific type by passing a regex
+     * @param folder contains the GIPP xml files
+     * @param a regex that match a specific type of GIPP
+     * @return the first occurrence that correspond to the input regex
+     */
+    public File findFile(File[] directories, String regexMatch) {
+        List<File> foundFiles = findFiles(directories, regexMatch);
+
+        if (foundFiles != null && foundFiles.size() > 0) {
+            return foundFiles.get(0);
+        }
+        else {
+            return null;
+        }
+    }
+
+    /**
+     * Constructor
+     * @param folder contains the GIPP xml files
+     */
+    public GIPPFileManager(String folder) {
+        File gippFolder = new File(folder);
+
+        if (!gippFolder.exists() || !gippFolder.isDirectory()) {
+            LOGGER.severe("The gipp folder " + gippFolder + " doesn't exist or is not a directory");
+        }
+
+        File[] directories = gippFolder.listFiles();
```

Comment:
Do we add the number of expected file per type og GIPP ? (if 2 SPAMOD available, we do not know which one to take, it should at least give a warning, if not en error)

Reply:
https://github.com/sen2vm/sen2vm-core/blob/62649dcfb05f9a10b2f160648f54796ba815e9f2/src/main/java/esa/sen2vm/input/gipp/GIPPManager.java#L141

Here we check that all asked bands are present in the product folder. If it's not the case, we throw an exception